modules/kafka/src/main/java/com/devonfw/module/kafka/common/messaging/api/Message.java

```diff
@@ -0,0 +1,195 @@
+package com.devonfw.module.kafka.common.messaging.api;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import com.devonfw.module.kafka.common.messaging.impl.MessageMetaData;
+
+/**
+ * This class is a message which needs to be send to kafkaBroker along with headers.
+ *
+ * @param <T> the type of Message.
+ */
+public class Message<T> {
+
+  private T payload;
+
+  private Map<String, String> customHeaders = new HashMap<>();
+
+  private Map<String, String> systemHeaders = new HashMap<>();
+
+  private Map<String, String> kafkaMetaData = new HashMap<>();
```

Comment:
yes , but this looked as very good implementation keeping our own message with lots of custom headers which is very important for eg. correlation id if suppose Tracer will get removed.

If its just ProducerRecord . it can be directly sent to KafkaTemplate instead of MessageSender class.

Reply:
But why it needs to be different types? What is a systemheader, what custom and what kafkametadata? I think if it possible to use kafkatemplate directly for special cases would be an advantage.

Reply:
i have removed all custom headers